baseapp/files/rest_framework/uploads/serializers.py

```
@@ -0,0 +1,94 @@
+    parent_id = serializers.UUIDField(required=False, allow_null=True)
+
+    def validate_parent_id(self, value):
+        """Validate parent_id exists as a DocumentId public_id."""
+        if not value:
+            return None
+
+        try:
+            document_id = DocumentId.objects.get(public_id=value)
+            return document_id.pk
+        except DocumentId.DoesNotExist:
+            raise serializers.ValidationError("Invalid parent_id: DocumentId not found")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)